### PR TITLE
chore(android_alarm_manager_plus): bump gradle and kotlin version

### PR DESCRIPTION
**Pull request by TheFinestArtist**
## Description

bump gradle version to 7.3.1, kotlin version to 1.7.22, and gradle-wrapper's gradle version to 7.5.1

## Description

*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this PR is changing it, and what motivated the change.*

## Checklist

- [x] I read the [Contributor Guide](https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I titled the PR using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0).
- [x] I did not modify the `CHANGELOG.md` nor the `pubspec.yaml` files.
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate that with a `!` in the title as explained in [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0)).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/android_alarm_manager_plus/example/android/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #Tue Oct 05 12:15:50 EEST 2021
 distributionBase=GRADLE_USER_HOME
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.2-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-all.zip
```
Comment (nohli) [inline packages/android_alarm_manager_plus/example/android/gradle/wrapper/gradle-wrapper.properties:3]: Would *-bin.zip be sufficient?

Comment (vbuberen) [reply packages/android_alarm_manager_plus/example/android/gradle/wrapper/gradle-wrapper.properties]: It is fine to do more commits as we are squashing them on merge.

**Review by vbuberen — COMMENTED** [review]
Thanks for you contribution, but we already have Dependabot taking care of such changes. We will merge this PR, but in future I would prefer Dependabot PRs.

**Comment by TheFinestArtist** [comment]
Sounds great to me. Next time, I will either wait for the bot to update dependencies or figure out how to use the bot.

**Review by vbuberen — APPROVED** [review]

**Pull request merged**